lualib/resty/events/codec.lua

```diff
@@ -0,0 +1,10 @@
+-- string.buffer is introduced since openresty 1.21.4.1
+
+local ok, buffer = pcall(require, "string.buffer")
+if not ok then
+    return require "cjson.safe"
+
+else
+    return buffer
```

Comment:
Discussed off-line, it seems that we should use separate buffer for encoding/decodng, as well as predefine the key names for faster and more compact serialization result.

Reply:
Great idea, I have add the new code for it.